silkworm/capi/snapshots.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+   Copyright 2025 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "snapshots.hpp"
+
+#include <silkworm/db/datastore/snapshots/snapshot_bundle.hpp>
+#include <silkworm/db/state/schema_config.hpp>
+
+using namespace silkworm;
+using namespace silkworm::db;
+using namespace silkworm::snapshots;
+
+MemoryMappedRegion make_region(const SilkwormMemoryMappedFile& mmf) {
+    return {mmf.memory_address, mmf.memory_length};
+}
+
+static std::optional<SnapshotPath> parse_snapshot_path(const char* file_path) {
+    if (!file_path) return std::nullopt;
+    return SnapshotPath::parse(file_path);
+}
+
+SnapshotBundleEntityDataResult build_inverted_index_entity_data(const SilkwormInvertedIndexSnapshot& snapshot) {
+    const auto ii_segment_path = parse_snapshot_path(snapshot.segment.file_path);
+    if (!ii_segment_path) {
+        return tl::make_unexpected(SILKWORM_INVALID_PATH);
+    }
+    segment::KVSegmentFileReader ii_segment{*ii_segment_path, seg::CompressionKind::kNone, make_region(snapshot.segment)};
+    const auto ii_accessor_index_path = parse_snapshot_path(snapshot.accessor_index.file_path);
+    if (!ii_accessor_index_path) {
+        return tl::make_unexpected(SILKWORM_INVALID_PATH);
+    }
+    rec_split::AccessorIndex ii_accessor_index{*ii_accessor_index_path, make_region(snapshot.accessor_index)};
+
+    SnapshotBundleEntityData data;
+    data.kv_segments.emplace(Schema::kInvIdxKVSegmentName, std::move(ii_segment));
+    data.accessor_indexes.emplace(Schema::kInvIdxAccessorIndexName, std::move(ii_accessor_index));
+    return data;
+}
+
+SnapshotBundleEntityDataResult build_domain_entity_data(const SilkwormDomainSnapshot& snapshot, uint32_t salt) {
+    // Domain snapshot files
+    const auto d_segment_path = parse_snapshot_path(snapshot.segment.file_path);
+    if (!d_segment_path) {
+        return tl::make_unexpected(SILKWORM_INVALID_PATH);
+    }
+    segment::KVSegmentFileReader d_segment{*d_segment_path, seg::CompressionKind::kAll, make_region(snapshot.segment)};
```

Comment:
Compression kind is different depending on a domain (see state/schema_config.cpp). Instead of hardcoding it in this file, I suggest to pass state_repository->schema() to the functions here, and obtain the right compression kind from there. It can be found by:
 
```
EntityDef& entity_def = schema.domain(name);
auto compression_kind = entity_def.kv_segment(Schema::kDomainKVSegmentName).compression_kind();
```

I suggest to do the same in build_inverted_index_entity_data for ii_segment for consistency, obtained from:
```
EntityDef& entity_def = schema.inverted_index(name);
auto compression_kind = entity_def.kv_segment(Schema::kInvIdxKVSegmentName).compression_kind();
```